source/server/api_listener_impl.h

```diff
@@ -0,0 +1,176 @@
+  // Server::Configuration::FactoryContext
+  AccessLog::AccessLogManager& accessLogManager() override;
+  Upstream::ClusterManager& clusterManager() override;
+  Event::Dispatcher& dispatcher() override;
+  Network::DrainDecision& drainDecision() override;
+  Grpc::Context& grpcContext() override;
+  bool healthCheckFailed() override;
+  Tracing::HttpTracer& httpTracer() override;
+  Http::Context& httpContext() override;
+  Init::Manager& initManager() override;
+  const LocalInfo::LocalInfo& localInfo() const override;
+  Envoy::Runtime::RandomGenerator& random() override;
+  Envoy::Runtime::Loader& runtime() override;
+  Stats::Scope& scope() override;
+  Singleton::Manager& singletonManager() override;
+  OverloadManager& overloadManager() override;
+  ThreadLocal::Instance& threadLocal() override;
+  Admin& admin() override;
+  const envoy::config::core::v3alpha::Metadata& listenerMetadata() const override;
+  envoy::config::core::v3alpha::TrafficDirection direction() const override;
+  TimeSource& timeSource() override;
+  ProtobufMessage::ValidationVisitor& messageValidationVisitor() override;
+  Api::Api& api() override;
+  ServerLifecycleNotifier& lifecycleNotifier() override;
+  OptProcessContextRef processContext() override;
+  Configuration::ServerFactoryContext& getServerFactoryContext() const override;
+  Stats::Scope& listenerScope() override;
+
+  // Network::DrainDecision
+  // TODO(junr03): hook up draining to listener state management.
+  bool drainClose() const override { return false; }
+
+private:
+  // Synthetic class that acts as a stub Network::ReadFilterCallbacks.
+  // TODO(junr03): if we are able to separate the Network Filter aspects of the
+  // Http::ConnectionManagerImpl from the http management aspects of it, it is possible we would not
+  // need this and the SyntheticConnection stub anymore.
+  class SyntheticReadCallbacks : public Network::ReadFilterCallbacks {
+  public:
+    SyntheticReadCallbacks(HttpApiListenerImpl& parent)
+        : parent_(parent), connection_(SyntheticConnection(*this)) {}
+
+    // Network::ReadFilterCallbacks
+    void continueReading() override {}
+    void injectReadDataToFilterChain(Buffer::Instance&, bool) override {}
+    Upstream::HostDescriptionConstSharedPtr upstreamHost() override { return nullptr; }
+    void upstreamHost(Upstream::HostDescriptionConstSharedPtr) override {}
+    Network::Connection& connection() override { return connection_; }
+
+    // Synthetic class that acts as a stub for the connection backing the
+    // Network::ReadFilterCallbacks.
+    class SyntheticConnection : public Network::Connection {
+    public:
+      SyntheticConnection(SyntheticReadCallbacks& parent)
+          : parent_(parent), stream_info_(parent_.parent_.timeSource()),
+            options_(std::make_shared<std::vector<Network::Socket::OptionConstSharedPtr>>()) {}
+
+      // Network::FilterManager
+      void addWriteFilter(Network::WriteFilterSharedPtr) override {}
```

Comment:
I would probably NOT_IMPLEMENTED all of these that you never expect to get called for clarity. Obviously this is a bit of an interface smell but I'm not sure there is much we can easily do about that.

Reply:
Yeah I thought about this for a bit, but without refactoring to separate the filter management aspect from the listener aspect per se, I didn't think we could do anything to clean up the interface.

I'll go ahead and replace these with NOT_IMPLEMENTED.